src/components/dialog-manager.js

```diff
@@ -73,15 +73,23 @@ export class DialogManager {
    * @return {!Promise}
    */
   openView(view, hidden = false) {
-    view.whenComplete().catch((reason) => {
+    this.handleView(view);
+    return this.openDialog(hidden).then((dialog) => {
+      return dialog.openView(view);
+    });
+  }
+
+  /**
+   * @param {!./view.View} view
+   * @return {!Promise}
+   */
+  handleView(view) {
```

Comment:
The `openView` method seemed pretty self-explanatory. What is the "handleView" method handling exactly?

Reply:
It'd be good to have a clearer name and comments for the method, if possible

Reply:
Changed function name and added comments.